<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>CITE-seq data with CiteFuse and MuData • MuData</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="CITE-seq data with CiteFuse and MuData">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MuData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.11.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Blood-CITE-seq.html">CITE-seq data with CiteFuse and MuData</a></li>
    <li><a class="dropdown-item" href="../articles/Cord-Blood-CITE-seq.html">CITE-seq data with MultiAssayExperiment and MuData</a></li>
    <li><a class="dropdown-item" href="../articles/Getting-Started.html">Getting started with MuData for MultiAssayExperiment</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ilia-kats/MuData/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>CITE-seq data with CiteFuse and MuData</h1>
                        <h4 data-toc-skip class="author">Danila
Bredikhin</h4>
            <address class="author_afil">
      European Molecular Biology Laboratory, Heidelberg,
Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:danila.bredikhin@embl.de" class="email">danila.bredikhin@embl.de</a>
      </address>
                              <h4 data-toc-skip class="author">Ilia
Kats</h4>
            <address class="author_afil">
      German Cancer Research Center, Heidelberg,
Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:i.kats@dkfz-heidelberg.de" class="email">i.kats@dkfz-heidelberg.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">2025-02-24</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ilia-kats/MuData/blob/master/vignettes/Blood-CITE-seq.Rmd" class="external-link"><code>vignettes/Blood-CITE-seq.Rmd</code></a></small>
      <div class="d-none name"><code>Blood-CITE-seq.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>CITE-seq data provide RNA and surface protein counts for the same
cells. This tutorial shows how MuData can be integrated into with
Bioconductor workflows to analyse CITE-seq data.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The most recent dev build can be installed from GitHub:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://remotes.r-lib.org" class="external-link">remotes</a></span><span class="op">)</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"ilia-kats/MuData"</span><span class="op">)</span></span></code></pre></div>
<p>Stable version of <code>MuData</code> will be available in future
bioconductor versions.</p>
</div>
<div class="section level2">
<h2 id="loading-libraries">Loading libraries<a class="anchor" aria-label="anchor" href="#loading-libraries"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ilia-kats/MuData" class="external-link">MuData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/" class="external-link">MultiAssayExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">CiteFuse</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/grimbough/rhdf5" class="external-link">rhdf5</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<p>We will use CITE-seq data available within <a href="http://www.bioconductor.org/packages/release/bioc/html/CiteFuse.html" class="external-link"><code>CiteFuse</code>
Bioconductor package</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CITEseq_example"</span>, package <span class="op">=</span> <span class="st">"CiteFuse"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">CITEseq_example</span>, <span class="va">dim</span><span class="op">)</span></span>
<span><span class="co">#&gt; $RNA</span></span>
<span><span class="co">#&gt; [1] 19521   500</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $ADT</span></span>
<span><span class="co">#&gt; [1]  49 500</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $HTO</span></span>
<span><span class="co">#&gt; [1]   4 500</span></span></code></pre></div>
<p>This dataset contains three matrices — one with <code>RNA</code>
counts, one with antibody-derived tags (<code>ADT</code>) counts and one
with hashtag oligonucleotide (<code>HTO</code>) counts.</p>
</div>
<div class="section level2">
<h2 id="processing-count-matrices">Processing count matrices<a class="anchor" aria-label="anchor" href="#processing-count-matrices"></a>
</h2>
<p>While CITE-seq analysis workflows such as <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/CiteFuse/inst/doc/CiteFuse.html" class="external-link">CiteFuse</a>
should be consulted for more details, below we exemplify simple data
transformations in order to demonstrate how their output can be saved to
an H5MU file later on.</p>
<p>Following the CiteFuse tutorial, we start with creating a
<code>SingleCellExperiment</code> object with the three matrices:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/preprocessing.html" class="external-link">preprocessing</a></span><span class="op">(</span><span class="va">CITEseq_example</span><span class="op">)</span></span>
<span><span class="va">sce_citeseq</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 19521 500 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(19521): hg19_AL627309.1 hg19_AL669831.5 ... hg19_MT-ND6</span></span>
<span><span class="co">#&gt;   hg19_MT-CYB</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(500): AAGCCGCGTTGTCTTT GATCGCGGTTATCGGT ... TTGGCAACACTAGTAC</span></span>
<span><span class="co">#&gt;   GCTGCGAGTTGTGGCC</span></span>
<span><span class="co">#&gt; colData names(0):</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(2): ADT HTO</span></span></code></pre></div>
<p>We will add a new assay with normalised RNA counts:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html" class="external-link">logNormCounts</a></span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span></span>
<span><span class="va">sce_citeseq</span>  <span class="co"># new assay: logcounts</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 19521 500 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span><span class="co">#&gt; rownames(19521): hg19_AL627309.1 hg19_AL669831.5 ... hg19_MT-ND6</span></span>
<span><span class="co">#&gt;   hg19_MT-CYB</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(500): AAGCCGCGTTGTCTTT GATCGCGGTTATCGGT ... TTGGCAACACTAGTAC</span></span>
<span><span class="co">#&gt;   GCTGCGAGTTGTGGCC</span></span>
<span><span class="co">#&gt; colData names(1): sizeFactor</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(2): ADT HTO</span></span></code></pre></div>
<p>To the ADT modality, we will add an assay with normalised counts:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu">CiteFuse</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/normaliseExprs.html" class="external-link">normaliseExprs</a></span><span class="op">(</span></span>
<span>  <span class="va">sce_citeseq</span>, altExp_name <span class="op">=</span> <span class="st">"ADT"</span>, transform <span class="op">=</span> <span class="st">"log"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, <span class="st">"ADT"</span><span class="op">)</span>  <span class="co"># new assay: logcounts</span></span>
<span><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 49 500 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span><span class="co">#&gt; rownames(49): B220 (CD45R) B7-H1 (PD-L1) ... TCRb TCRg</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(500): AAGCCGCGTTGTCTTT GATCGCGGTTATCGGT ... TTGGCAACACTAGTAC</span></span>
<span><span class="co">#&gt;   GCTGCGAGTTGTGGCC</span></span>
<span><span class="co">#&gt; colData names(0):</span></span></code></pre></div>
<p>We will also generate reduced dimensions:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span></span>
<span>  <span class="va">sce_citeseq</span>, exprs_values <span class="op">=</span> <span class="st">"logcounts"</span>, ncomponents <span class="op">=</span> <span class="fl">20</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html" class="external-link">plotReducedDim</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, dimred <span class="op">=</span> <span class="st">"PCA"</span>, </span>
<span>                       by_exprs_values <span class="op">=</span> <span class="st">"logcounts"</span>, colour_by <span class="op">=</span> <span class="st">"CD27"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Blood-CITE-seq_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="making-a-multiassayexperiment-object">Making a MultiAssayExperiment object<a class="anchor" aria-label="anchor" href="#making-a-multiassayexperiment-object"></a>
</h2>
<p>An appropriate structure for multimodal datasets is <a href="https://bioconductor.org/packages/MultiAssayExperiment/" class="external-link"><code>MultiAssayExperiment</code></a>.</p>
<p>We will make a respective MultiAssayExperiment object from
<code>sce_citeseq</code>:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">experiments</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  ADT <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, <span class="st">"ADT"</span><span class="op">)</span>,</span>
<span>  HTO <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, <span class="st">"HTO"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Drop other modalities from sce_citeseq</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">experiments</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">sce_citeseq</span></span>
<span></span>
<span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://github.com/waldronlab/MultiAssayExperiment/reference/MultiAssayExperiment.html" class="external-link">MultiAssayExperiment</a></span><span class="op">(</span><span class="va">experiments</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="writing-to-h5mu">Writing to H5MU<a class="anchor" aria-label="anchor" href="#writing-to-h5mu"></a>
</h2>
<p>We can write the contents of the MultiAssayExperiment object into an
H5MU file:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/writeH5MU.html">writeH5MU</a></span><span class="op">(</span><span class="va">mae</span>, <span class="st">"citefuse_example.h5mu"</span><span class="op">)</span></span></code></pre></div>
<p>We can check that all the modalities were written to the file:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h5</span> <span class="op">&lt;-</span> <span class="fu">rhdf5</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Fopen.html" class="external-link">H5Fopen</a></span><span class="op">(</span><span class="st">"citefuse_example.h5mu"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html" class="external-link">h5ls</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Gopen.html" class="external-link">H5Gopen</a></span><span class="op">(</span><span class="va">h5</span>, <span class="st">"mod"</span><span class="op">)</span>, recursive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt;   group name     otype dclass dim</span></span>
<span><span class="co">#&gt; 0     /  ADT H5I_GROUP           </span></span>
<span><span class="co">#&gt; 1     /  HTO H5I_GROUP           </span></span>
<span><span class="co">#&gt; 2     /  RNA H5I_GROUP</span></span></code></pre></div>
<p>… both assays for ADT — raw counts are stored in <code>X</code> and
normalised counts are in the corresponding layer:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html" class="external-link">h5ls</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Gopen.html" class="external-link">H5Gopen</a></span><span class="op">(</span><span class="va">h5</span>, <span class="st">"mod/ADT"</span><span class="op">)</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt;   group   name     otype dclass dim</span></span>
<span><span class="co">#&gt; 0     /      X H5I_GROUP           </span></span>
<span><span class="co">#&gt; 1     / layers H5I_GROUP           </span></span>
<span><span class="co">#&gt; 2     /    obs H5I_GROUP           </span></span>
<span><span class="co">#&gt; 3     /    var H5I_GROUP</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html" class="external-link">h5ls</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Gopen.html" class="external-link">H5Gopen</a></span><span class="op">(</span><span class="va">h5</span>, <span class="st">"mod/ADT/layers"</span><span class="op">)</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt;   group      name       otype dclass      dim</span></span>
<span><span class="co">#&gt; 0     / logcounts H5I_DATASET  FLOAT 49 x 500</span></span></code></pre></div>
<p>… as well as reduced dimensions (PCA):</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html" class="external-link">h5ls</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Gopen.html" class="external-link">H5Gopen</a></span><span class="op">(</span><span class="va">h5</span>, <span class="st">"mod/RNA/obsm"</span><span class="op">)</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt;   group name       otype dclass      dim</span></span>
<span><span class="co">#&gt; 0     /  PCA H5I_DATASET  FLOAT 20 x 500</span></span>
<span><span class="co"># There is an alternative way to access groups:</span></span>
<span><span class="co"># h5&amp;'mod'&amp;'RNA'&amp;'obsm'</span></span>
<span><span class="fu">rhdf5</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5functions.html" class="external-link">H5close</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li><p><a href="https://www.biorxiv.org/content/10.1101/2021.06.01.445670" class="external-link">Muon:
multimodal omics analysis framework</a> preprint</p></li>
<li><p><a href="https://mudata.readthedocs.io/" class="external-link">mudata</a> (Python)
documentation</p></li>
<li><p>muon <a href="https://muon.readthedocs.io/" class="external-link">documentation</a> and
<a href="https://gtca.github.io/muon/" class="external-link">web page</a></p></li>
<li><p>Kim HJ, Lin Y, Geddes TA, Yang P, Yang JYH (2020). “CiteFuse
enables multi-modal analysis of CITE-seq data.” Bioinformatics, 36(14),
4137–4143. <a href="https://doi.org/10.1093/bioinformatics/btaa282" class="external-link uri">https://doi.org/10.1093/bioinformatics/btaa282</a>.</p></li>
<li><p>Ramos M, Schiffer L, Re A, Azhar R, Basunia A, Cabrera CR, Chan
T, Chapman P, Davis S, Gomez-Cabrero D, Culhane AC, Haibe-Kains B,
Hansen K, Kodali H, Louis MS, Mer AS, Reister M, Morgan M, Carey V,
Waldron L (2017). “Software For The Integration Of Multi-Omics
Experiments In Bioconductor.” Cancer Research, 77(21); e39-42.</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R Under development (unstable) (2025-02-22 r87796)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.1 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] scater_1.35.1               ggplot2_3.5.1              </span></span>
<span><span class="co">#&gt;  [3] scuttle_1.17.0              CiteFuse_1.19.0            </span></span>
<span><span class="co">#&gt;  [5] MultiAssayExperiment_1.33.9 SingleCellExperiment_1.29.1</span></span>
<span><span class="co">#&gt;  [7] SummarizedExperiment_1.37.0 Biobase_2.67.0             </span></span>
<span><span class="co">#&gt;  [9] GenomicRanges_1.59.1        GenomeInfoDb_1.43.4        </span></span>
<span><span class="co">#&gt; [11] IRanges_2.41.3              MatrixGenerics_1.19.1      </span></span>
<span><span class="co">#&gt; [13] matrixStats_1.5.0           MuData_1.11.1              </span></span>
<span><span class="co">#&gt; [15] rhdf5_2.51.2                S4Vectors_0.45.4           </span></span>
<span><span class="co">#&gt; [17] BiocGenerics_0.53.6         generics_0.1.3             </span></span>
<span><span class="co">#&gt; [19] Matrix_1.7-2                BiocStyle_2.35.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3      tensorA_0.36.2.1        jsonlite_1.9.0         </span></span>
<span><span class="co">#&gt;   [4] magrittr_2.0.3          ggbeeswarm_0.7.2        farver_2.1.2           </span></span>
<span><span class="co">#&gt;   [7] rmarkdown_2.29          fs_1.6.5                ragg_1.3.3             </span></span>
<span><span class="co">#&gt;  [10] vctrs_0.6.5             memoise_2.0.1           htmltools_0.5.8.1      </span></span>
<span><span class="co">#&gt;  [13] S4Arrays_1.7.3          BiocBaseUtils_1.9.0     BiocNeighbors_2.1.2    </span></span>
<span><span class="co">#&gt;  [16] Rhdf5lib_1.29.0         SparseArray_1.7.6       sass_0.4.9             </span></span>
<span><span class="co">#&gt;  [19] bslib_0.9.0             htmlwidgets_1.6.4       desc_1.4.3             </span></span>
<span><span class="co">#&gt;  [22] plyr_1.8.9              plotly_4.10.4           cachem_1.1.0           </span></span>
<span><span class="co">#&gt;  [25] igraph_2.1.4            lifecycle_1.0.4         pkgconfig_2.0.3        </span></span>
<span><span class="co">#&gt;  [28] rsvd_1.0.5              R6_2.6.1                fastmap_1.2.0          </span></span>
<span><span class="co">#&gt;  [31] GenomeInfoDbData_1.2.13 digest_0.6.37           colorspace_2.1-1       </span></span>
<span><span class="co">#&gt;  [34] dqrng_0.4.1             irlba_2.3.5.1           textshaping_1.0.0      </span></span>
<span><span class="co">#&gt;  [37] beachmat_2.23.6         labeling_0.4.3          randomForest_4.7-1.2   </span></span>
<span><span class="co">#&gt;  [40] httr_1.4.7              polyclip_1.10-7         abind_1.4-8            </span></span>
<span><span class="co">#&gt;  [43] compiler_4.5.0          withr_3.0.2             BiocParallel_1.41.2    </span></span>
<span><span class="co">#&gt;  [46] viridis_0.6.5           ggforce_0.4.2           MASS_7.3-64            </span></span>
<span><span class="co">#&gt;  [49] bayesm_3.1-6            DelayedArray_0.33.6     bluster_1.17.0         </span></span>
<span><span class="co">#&gt;  [52] tools_4.5.0             vipor_0.4.7             beeswarm_0.4.0         </span></span>
<span><span class="co">#&gt;  [55] glue_1.8.0              dbscan_1.2.2            nlme_3.1-167           </span></span>
<span><span class="co">#&gt;  [58] rhdf5filters_1.19.1     grid_4.5.0              Rtsne_0.17             </span></span>
<span><span class="co">#&gt;  [61] cluster_2.1.8           reshape2_1.4.4          gtable_0.3.6           </span></span>
<span><span class="co">#&gt;  [64] tidyr_1.3.1             data.table_1.17.0       BiocSingular_1.23.0    </span></span>
<span><span class="co">#&gt;  [67] tidygraph_1.3.1         ScaledMatrix_1.15.0     metapod_1.15.0         </span></span>
<span><span class="co">#&gt;  [70] XVector_0.47.2          ggrepel_0.9.6           pillar_1.10.1          </span></span>
<span><span class="co">#&gt;  [73] stringr_1.5.1           limma_3.63.3            robustbase_0.99-4-1    </span></span>
<span><span class="co">#&gt;  [76] splines_4.5.0           dplyr_1.1.4             tweenr_2.0.3           </span></span>
<span><span class="co">#&gt;  [79] lattice_0.22-6          survival_3.8-3          compositions_2.0-8     </span></span>
<span><span class="co">#&gt;  [82] tidyselect_1.2.1        locfit_1.5-9.11         knitr_1.49             </span></span>
<span><span class="co">#&gt;  [85] gridExtra_2.3           bookdown_0.42           edgeR_4.5.2            </span></span>
<span><span class="co">#&gt;  [88] xfun_0.51               graphlayouts_1.2.2      mixtools_2.0.0         </span></span>
<span><span class="co">#&gt;  [91] statmod_1.5.0           DEoptimR_1.1-3-1        pheatmap_1.0.12        </span></span>
<span><span class="co">#&gt;  [94] stringi_1.8.4           UCSC.utils_1.3.1        lazyeval_0.2.2         </span></span>
<span><span class="co">#&gt;  [97] yaml_2.3.10             evaluate_1.0.3          codetools_0.2-20       </span></span>
<span><span class="co">#&gt; [100] kernlab_0.9-33          ggraph_2.2.1            tibble_3.2.1           </span></span>
<span><span class="co">#&gt; [103] BiocManager_1.30.25     cli_3.6.4               uwot_0.2.3             </span></span>
<span><span class="co">#&gt; [106] systemfonts_1.2.1       segmented_2.1-3         munsell_0.5.1          </span></span>
<span><span class="co">#&gt; [109] jquerylib_0.1.4         Rcpp_1.0.14             parallel_4.5.0         </span></span>
<span><span class="co">#&gt; [112] pkgdown_2.1.1           scran_1.35.0            viridisLite_0.4.2      </span></span>
<span><span class="co">#&gt; [115] scales_1.3.0            ggridges_0.5.6          purrr_1.0.4            </span></span>
<span><span class="co">#&gt; [118] crayon_1.5.3            rlang_1.1.5             cowplot_1.1.3</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Danila Bredikhin, Ilia Kats.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
