<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>CITE-seq data with MultiAssayExperiment and MuData â€¢ MuData</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="CITE-seq data with MultiAssayExperiment and MuData">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MuData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.11.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Blood-CITE-seq.html">CITE-seq data with CiteFuse and MuData</a></li>
    <li><a class="dropdown-item" href="../articles/Cord-Blood-CITE-seq.html">CITE-seq data with MultiAssayExperiment and MuData</a></li>
    <li><a class="dropdown-item" href="../articles/Getting-Started.html">Getting started with MuData for MultiAssayExperiment</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ilia-kats/MuData/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>CITE-seq data with MultiAssayExperiment and MuData</h1>
                        <h4 data-toc-skip class="author">Danila
Bredikhin</h4>
            <address class="author_afil">
      European Molecular Biology Laboratory, Heidelberg,
Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:danila.bredikhin@embl.de" class="email">danila.bredikhin@embl.de</a>
      </address>
                              <h4 data-toc-skip class="author">Ilia
Kats</h4>
            <address class="author_afil">
      German Cancer Research Center, Heidelberg,
Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:i.kats@dkfz-heidelberg.de" class="email">i.kats@dkfz-heidelberg.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">2025-02-24</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ilia-kats/MuData/blob/master/vignettes/Cord-Blood-CITE-seq.Rmd" class="external-link"><code>vignettes/Cord-Blood-CITE-seq.Rmd</code></a></small>
      <div class="d-none name"><code>Cord-Blood-CITE-seq.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>CITE-seq data provide RNA and surface protein counts for the same
cells. This tutorial shows how MuData can be integrated into with
Bioconductor workflows to analyse CITE-seq data.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The most recent dev build can be installed from GitHub:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://remotes.r-lib.org" class="external-link">remotes</a></span><span class="op">)</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"ilia-kats/MuData"</span><span class="op">)</span></span></code></pre></div>
<p>Stable version of <code>MuData</code> will be available in future
bioconductor versions.</p>
</div>
<div class="section level2">
<h2 id="loading-libraries">Loading libraries<a class="anchor" aria-label="anchor" href="#loading-libraries"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ilia-kats/MuData" class="external-link">MuData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/" class="external-link">MultiAssayExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellMultiModal</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/grimbough/rhdf5" class="external-link">rhdf5</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<p>We will use CITE-seq data accessible with the <a href="https://bioconductor.org/packages/release/data/experiment/vignettes/SingleCellMultiModal/inst/doc/CITEseq.html" class="external-link"><code>SingleCellMultiModal</code>
Bioconductor package</a>, which was originally described in <a href="https://www.nature.com/articles/nmeth.4380" class="external-link">Stoeckius et al.,
2017</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellMultiModal/man/CITEseq.html" class="external-link">CITEseq</a></span><span class="op">(</span></span>
<span>    DataType<span class="op">=</span><span class="st">"cord_blood"</span>, modes<span class="op">=</span><span class="st">"*"</span>, dry.run<span class="op">=</span><span class="cn">FALSE</span>, version<span class="op">=</span><span class="st">"1.0.0"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Dataset: cord_blood</span></span>
<span><span class="co">#&gt; Working on: scADT_Counts</span></span>
<span><span class="co">#&gt; Working on: scRNAseq_Counts</span></span>
<span><span class="co">#&gt; Working on: coldata_scRNAseq</span></span>
<span><span class="co">#&gt; Working on: scADT_clrCounts</span></span>
<span><span class="co">#&gt; see ?SingleCellMultiModal and browseVignettes('SingleCellMultiModal') for documentation</span></span>
<span><span class="co">#&gt; downloading 1 resources</span></span>
<span><span class="co">#&gt; retrieving 1 resource</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; see ?SingleCellMultiModal and browseVignettes('SingleCellMultiModal') for documentation</span></span>
<span><span class="co">#&gt; downloading 1 resources</span></span>
<span><span class="co">#&gt; retrieving 1 resource</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; see ?SingleCellMultiModal and browseVignettes('SingleCellMultiModal') for documentation</span></span>
<span><span class="co">#&gt; downloading 1 resources</span></span>
<span><span class="co">#&gt; retrieving 1 resource</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; see ?SingleCellMultiModal and browseVignettes('SingleCellMultiModal') for documentation</span></span>
<span><span class="co">#&gt; downloading 1 resources</span></span>
<span><span class="co">#&gt; retrieving 1 resource</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; Warning: 'ExperimentList' contains 'data.frame' or 'DataFrame',</span></span>
<span><span class="co">#&gt;   potential for errors with mixed data types</span></span>
<span><span class="co">#&gt; harmonizing input:</span></span>
<span><span class="co">#&gt;   removing 2277 sampleMap rows with 'primary' not in colData</span></span>
<span></span>
<span><span class="va">mae</span></span>
<span><span class="co">#&gt; A MultiAssayExperiment object of 3 listed</span></span>
<span><span class="co">#&gt;  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">#&gt;  Containing an ExperimentList class object of length 3:</span></span>
<span><span class="co">#&gt;  [1] scADT_clr: matrix with 13 rows and 7858 columns</span></span>
<span><span class="co">#&gt;  [2] scADT: matrix with 13 rows and 7858 columns</span></span>
<span><span class="co">#&gt;  [3] scRNAseq: matrix with 36280 rows and 7858 columns</span></span>
<span><span class="co">#&gt; Functionality:</span></span>
<span><span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">#&gt;  exportClass() - save data to flat files</span></span></code></pre></div>
<p>We see two modalities in the object â€” <code>scRNAseq</code> and
<code>scADT</code>, the latter providing counts for antibody-derived
tags. Notably, each experiment is a matrix.</p>
</div>
<div class="section level2">
<h2 id="processing-adt-data">Processing ADT data<a class="anchor" aria-label="anchor" href="#processing-adt-data"></a>
</h2>
<p>While CITE-seq analysis workflows such as <a href="http://www.bioconductor.org/packages/release/bioc/html/CiteFuse.html" class="external-link">CiteFuse</a>
should be consulted for more details, below we exemplify simple data
transformation in order to demonstrate how their output can be saved to
an H5MU file later on.</p>
<p>For ADT counts, we will apply CLR transformation following <a href="https://doi.org/10.1016/j.cell.2021.04.048" class="external-link">Hao et al.,
2020</a>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define CLR transformation as in the Seurat workflow</span></span>
<span><span class="va">clr</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">data</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span></span>
<span>    <span class="va">x</span> <span class="op">/</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="va">x</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span><span class="op">)</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>We will make the ADT modality a <code>SingleCellExperiment</code>
object and add an assay with CLR-transformed counts:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">adt_counts</span> <span class="op">&lt;-</span> <span class="va">mae</span><span class="op">[[</span><span class="st">"scADT"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="va">mae</span><span class="op">[[</span><span class="st">"scADT"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a></span><span class="op">(</span><span class="va">adt_counts</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="st">"scADT"</span><span class="op">]</span><span class="op">]</span>, <span class="st">"clr"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">clr</span><span class="op">(</span><span class="va">adt_counts</span><span class="op">)</span></span></code></pre></div>
<p>We will also generate reduced dimensions taking advantage of the
functionality in the <a href="https://bioconductor.org/packages/release/bioc/vignettes/scater/inst/doc/overview.html" class="external-link"><code>scater</code>
package</a>:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span><span class="op">[[</span><span class="st">"scADT"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span></span>
<span>  <span class="va">mae</span><span class="op">[[</span><span class="st">"scADT"</span><span class="op">]</span><span class="op">]</span>, exprs_values <span class="op">=</span> <span class="st">"clr"</span>, ncomponents <span class="op">=</span> <span class="fl">20</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in check_numbers(k = k, nu = nu, nv = nv, limit = min(dim(x)) - : more</span></span>
<span><span class="co">#&gt; singular values/vectors requested than available</span></span>
<span><span class="co">#&gt; Warning in (function (A, nv = 5, nu = nv, maxit = 1000, work = nv + 7, reorth =</span></span>
<span><span class="co">#&gt; TRUE, : You're computing too large a percentage of total singular values, use a</span></span>
<span><span class="co">#&gt; standard svd instead.</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html" class="external-link">plotReducedDim</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="st">"scADT"</span><span class="op">]</span><span class="op">]</span>, dimred <span class="op">=</span> <span class="st">"PCA"</span>,</span>
<span>               by_exprs_values <span class="op">=</span> <span class="st">"clr"</span>, colour_by <span class="op">=</span> <span class="st">"CD3"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Cord-Blood-CITE-seq_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html" class="external-link">plotReducedDim</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="st">"scADT"</span><span class="op">]</span><span class="op">]</span>, dimred <span class="op">=</span> <span class="st">"PCA"</span>,</span>
<span>               by_exprs_values <span class="op">=</span> <span class="st">"clr"</span>, colour_by <span class="op">=</span> <span class="st">"CD14"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Cord-Blood-CITE-seq_files/figure-html/unnamed-chunk-7-2.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="writing-h5mu-files">Writing H5MU files<a class="anchor" aria-label="anchor" href="#writing-h5mu-files"></a>
</h2>
<p>We can write the contents of the MultiAssayExperiment object into an
H5MU file:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/writeH5MU.html">writeH5MU</a></span><span class="op">(</span><span class="va">mae</span>, <span class="st">"cord_blood_citeseq.h5mu"</span><span class="op">)</span></span></code></pre></div>
<p>We can check that both modalities were written to the file, whether
it was a <code>matrix</code> for RNA or
<code>SingleCellExperiment</code> for ADT:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h5</span> <span class="op">&lt;-</span> <span class="fu">rhdf5</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Fopen.html" class="external-link">H5Fopen</a></span><span class="op">(</span><span class="st">"cord_blood_citeseq.h5mu"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html" class="external-link">h5ls</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Gopen.html" class="external-link">H5Gopen</a></span><span class="op">(</span><span class="va">h5</span>, <span class="st">"mod"</span><span class="op">)</span>, recursive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt;   group      name     otype dclass dim</span></span>
<span><span class="co">#&gt; 0     /     scADT H5I_GROUP           </span></span>
<span><span class="co">#&gt; 1     / scADT_clr H5I_GROUP           </span></span>
<span><span class="co">#&gt; 2     /  scRNAseq H5I_GROUP</span></span></code></pre></div>
<p>â€¦ both assays for ADT â€” raw counts are stored in <code>X</code> and
CLR-transformed counts are in the corresponding layer:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html" class="external-link">h5ls</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Gopen.html" class="external-link">H5Gopen</a></span><span class="op">(</span><span class="va">h5</span>, <span class="st">"mod/scADT"</span><span class="op">)</span>, recursive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt;   group   name       otype  dclass       dim</span></span>
<span><span class="co">#&gt; 0     /      X H5I_DATASET INTEGER 13 x 7858</span></span>
<span><span class="co">#&gt; 1     / layers   H5I_GROUP                  </span></span>
<span><span class="co">#&gt; 2     /    obs   H5I_GROUP                  </span></span>
<span><span class="co">#&gt; 3     /   obsm   H5I_GROUP                  </span></span>
<span><span class="co">#&gt; 4     /    var   H5I_GROUP</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html" class="external-link">h5ls</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Gopen.html" class="external-link">H5Gopen</a></span><span class="op">(</span><span class="va">h5</span>, <span class="st">"mod/scADT/layers"</span><span class="op">)</span>, recursive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt;   group name       otype dclass       dim</span></span>
<span><span class="co">#&gt; 0     /  clr H5I_DATASET  FLOAT 13 x 7858</span></span></code></pre></div>
<p>â€¦ as well as reduced dimensions (PCA):</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html" class="external-link">h5ls</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Gopen.html" class="external-link">H5Gopen</a></span><span class="op">(</span><span class="va">h5</span>, <span class="st">"mod/scADT/obsm"</span><span class="op">)</span>, recursive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt;   group name       otype dclass       dim</span></span>
<span><span class="co">#&gt; 0     /  PCA H5I_DATASET  FLOAT 12 x 7858</span></span>
<span><span class="co"># There is an alternative way to access groups:</span></span>
<span><span class="co"># h5&amp;'mod'&amp;'scADT'&amp;'obsm'</span></span>
<span><span class="fu">rhdf5</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5functions.html" class="external-link">H5close</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li><p><a href="https://www.biorxiv.org/content/10.1101/2021.06.01.445670" class="external-link">Muon:
multimodal omics analysis framework</a> preprint</p></li>
<li><p><a href="https://mudata.readthedocs.io/" class="external-link">mudata</a> (Python)
documentation</p></li>
<li><p>muon <a href="https://muon.readthedocs.io/" class="external-link">documentation</a> and
<a href="https://gtca.github.io/muon/" class="external-link">web page</a></p></li>
<li><p>Stoeckius, M., Hafemeister, C., Stephenson, W., Houck-Loomis, B.,
Chattopadhyay, P.K., Swerdlow, H., Satija, R. and Smibert, P., 2017.
Simultaneous epitope and transcriptome measurement in single cells.
<em>Nature methods</em>, 14(9), pp.865-868.</p></li>
<li><p>Hao, Y., Hao, S., Andersen-Nissen, E., Mauck III, W.M., Zheng,
S., Butler, A., Lee, M.J., Wilk, A.J., Darby, C., Zager, M. and Hoffman,
P., 2021. Integrated analysis of multimodal single-cell data.
<em>Cell</em>.</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R Under development (unstable) (2025-02-22 r87796)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.1 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] scater_1.35.1               ggplot2_3.5.1              </span></span>
<span><span class="co">#&gt;  [3] scuttle_1.17.0              SingleCellMultiModal_1.19.1</span></span>
<span><span class="co">#&gt;  [5] MultiAssayExperiment_1.33.9 SingleCellExperiment_1.29.1</span></span>
<span><span class="co">#&gt;  [7] SummarizedExperiment_1.37.0 Biobase_2.67.0             </span></span>
<span><span class="co">#&gt;  [9] GenomicRanges_1.59.1        GenomeInfoDb_1.43.4        </span></span>
<span><span class="co">#&gt; [11] IRanges_2.41.3              MatrixGenerics_1.19.1      </span></span>
<span><span class="co">#&gt; [13] matrixStats_1.5.0           MuData_1.11.1              </span></span>
<span><span class="co">#&gt; [15] rhdf5_2.51.2                S4Vectors_0.45.4           </span></span>
<span><span class="co">#&gt; [17] BiocGenerics_0.53.6         generics_0.1.3             </span></span>
<span><span class="co">#&gt; [19] Matrix_1.7-2                BiocStyle_2.35.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] DBI_1.2.3                gridExtra_2.3            rlang_1.1.5             </span></span>
<span><span class="co">#&gt;  [4] magrittr_2.0.3           compiler_4.5.0           RSQLite_2.3.9           </span></span>
<span><span class="co">#&gt;  [7] png_0.1-8                systemfonts_1.2.1        vctrs_0.6.5             </span></span>
<span><span class="co">#&gt; [10] pkgconfig_2.0.3          SpatialExperiment_1.17.0 crayon_1.5.3            </span></span>
<span><span class="co">#&gt; [13] fastmap_1.2.0            dbplyr_2.5.0             magick_2.8.5            </span></span>
<span><span class="co">#&gt; [16] XVector_0.47.2           labeling_0.4.3           rmarkdown_2.29          </span></span>
<span><span class="co">#&gt; [19] ggbeeswarm_0.7.2         UCSC.utils_1.3.1         ragg_1.3.3              </span></span>
<span><span class="co">#&gt; [22] purrr_1.0.4              bit_4.5.0.1              xfun_0.51               </span></span>
<span><span class="co">#&gt; [25] cachem_1.1.0             beachmat_2.23.6          jsonlite_1.9.0          </span></span>
<span><span class="co">#&gt; [28] blob_1.2.4               rhdf5filters_1.19.1      DelayedArray_0.33.6     </span></span>
<span><span class="co">#&gt; [31] Rhdf5lib_1.29.0          BiocParallel_1.41.2      irlba_2.3.5.1           </span></span>
<span><span class="co">#&gt; [34] parallel_4.5.0           R6_2.6.1                 bslib_0.9.0             </span></span>
<span><span class="co">#&gt; [37] jquerylib_0.1.4          Rcpp_1.0.14              bookdown_0.42           </span></span>
<span><span class="co">#&gt; [40] knitr_1.49               BiocBaseUtils_1.9.0      tidyselect_1.2.1        </span></span>
<span><span class="co">#&gt; [43] viridis_0.6.5            abind_1.4-8              yaml_2.3.10             </span></span>
<span><span class="co">#&gt; [46] codetools_0.2-20         curl_6.2.1               lattice_0.22-6          </span></span>
<span><span class="co">#&gt; [49] tibble_3.2.1             withr_3.0.2              KEGGREST_1.47.0         </span></span>
<span><span class="co">#&gt; [52] evaluate_1.0.3           desc_1.4.3               BiocFileCache_2.15.1    </span></span>
<span><span class="co">#&gt; [55] ExperimentHub_2.15.0     Biostrings_2.75.4        pillar_1.10.1           </span></span>
<span><span class="co">#&gt; [58] BiocManager_1.30.25      filelock_1.0.3           BiocVersion_3.21.1      </span></span>
<span><span class="co">#&gt; [61] munsell_0.5.1            scales_1.3.0             glue_1.8.0              </span></span>
<span><span class="co">#&gt; [64] tools_4.5.0              AnnotationHub_3.15.0     BiocNeighbors_2.1.2     </span></span>
<span><span class="co">#&gt; [67] ScaledMatrix_1.15.0      fs_1.6.5                 cowplot_1.1.3           </span></span>
<span><span class="co">#&gt; [70] grid_4.5.0               AnnotationDbi_1.69.0     colorspace_2.1-1        </span></span>
<span><span class="co">#&gt; [73] GenomeInfoDbData_1.2.13  beeswarm_0.4.0           BiocSingular_1.23.0     </span></span>
<span><span class="co">#&gt; [76] vipor_0.4.7              rsvd_1.0.5               cli_3.6.4               </span></span>
<span><span class="co">#&gt; [79] rappdirs_0.3.3           textshaping_1.0.0        viridisLite_0.4.2       </span></span>
<span><span class="co">#&gt; [82] S4Arrays_1.7.3           dplyr_1.1.4              gtable_0.3.6            </span></span>
<span><span class="co">#&gt; [85] sass_0.4.9               digest_0.6.37            ggrepel_0.9.6           </span></span>
<span><span class="co">#&gt; [88] SparseArray_1.7.6        farver_2.1.2             rjson_0.2.23            </span></span>
<span><span class="co">#&gt; [91] htmlwidgets_1.6.4        memoise_2.0.1            htmltools_0.5.8.1       </span></span>
<span><span class="co">#&gt; [94] pkgdown_2.1.1            lifecycle_1.0.4          httr_1.4.7              </span></span>
<span><span class="co">#&gt; [97] mime_0.12                bit64_4.6.0-1</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Danila Bredikhin, Ilia Kats.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
