<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MuData">
<title>CITE-seq data with MultiAssayExperiment and MuData • MuData</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="CITE-seq data with MultiAssayExperiment and MuData">
<meta property="og:description" content="MuData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MuData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.9</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Blood-CITE-seq.html">CITE-seq data with CiteFuse and MuData</a>
    <a class="dropdown-item" href="../articles/Cord-Blood-CITE-seq.html">CITE-seq data with MultiAssayExperiment and MuData</a>
    <a class="dropdown-item" href="../articles/Getting-Started.html">Getting started with MuData for MultiAssayExperiment</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ilia-kats/MuData/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="Cord-Blood-CITE-seq_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>CITE-seq data with MultiAssayExperiment and MuData</h1>
                        <h4 data-toc-skip class="author">Danila Bredikhin</h4>
            <address class="author_afil">
      European Molecular Biology Laboratory, Heidelberg, Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:danila.bredikhin@embl.de" class="email">danila.bredikhin@embl.de</a>
      </address>
                              <h4 data-toc-skip class="author">Ilia Kats</h4>
            <address class="author_afil">
      German Cancer Research Center, Heidelberg, Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:i.kats@dkfz-heidelberg.de" class="email">i.kats@dkfz-heidelberg.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">2022-05-31</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ilia-kats/MuData/blob/HEAD/vignettes/Cord-Blood-CITE-seq.Rmd" class="external-link"><code>vignettes/Cord-Blood-CITE-seq.Rmd</code></a></small>
      <div class="d-none name"><code>Cord-Blood-CITE-seq.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>CITE-seq data provide RNA and surface protein counts for the same cells. This tutorial shows how MuData can be integrated into with Bioconductor workflows to analyse CITE-seq data.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The most recent dev build can be installed from GitHub:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://remotes.r-lib.org" class="external-link">remotes</a></span><span class="op">)</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"ilia-kats/MuData"</span><span class="op">)</span></code></pre></div>
<p>Stable version of <code>MuData</code> will be available in future bioconductor versions.</p>
</div>
<div class="section level2">
<h2 id="loading-libraries">Loading libraries<a class="anchor" aria-label="anchor" href="#loading-libraries"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ilia-kats/MuData" class="external-link">MuData</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/" class="external-link">MultiAssayExperiment</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellMultiModal</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/grimbough/rhdf5" class="external-link">rhdf5</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<p>We will use CITE-seq data accessible with the <a href="https://bioconductor.org/packages/release/data/experiment/vignettes/SingleCellMultiModal/inst/doc/CITEseq.html" class="external-link"><code>SingleCellMultiModal</code> Bioconductor package</a>, which was originally described in <a href="https://www.nature.com/articles/nmeth.4380" class="external-link">Stoeckius et al., 2017</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellMultiModal/man/CITEseq.html" class="external-link">CITEseq</a></span><span class="op">(</span>
    DataType<span class="op">=</span><span class="st">"cord_blood"</span>, modes<span class="op">=</span><span class="st">"*"</span>, dry.run<span class="op">=</span><span class="cn">FALSE</span>, version<span class="op">=</span><span class="st">"1.0.0"</span>
<span class="op">)</span>
<span class="co">#&gt; Dataset: cord_blood</span>
<span class="co">#&gt; snapshotDate(): 2022-05-17</span>
<span class="co">#&gt; Working on: scADT_Counts</span>
<span class="co">#&gt; Working on: scRNAseq_Counts</span>
<span class="co">#&gt; see ?SingleCellMultiModal and browseVignettes('SingleCellMultiModal') for documentation</span>
<span class="co">#&gt; downloading 1 resources</span>
<span class="co">#&gt; retrieving 1 resource</span>
<span class="co">#&gt; loading from cache</span>
<span class="co">#&gt; see ?SingleCellMultiModal and browseVignettes('SingleCellMultiModal') for documentation</span>
<span class="co">#&gt; downloading 1 resources</span>
<span class="co">#&gt; retrieving 1 resource</span>
<span class="co">#&gt; loading from cache</span>

<span class="va">mae</span>
<span class="co">#&gt; A MultiAssayExperiment object of 2 listed</span>
<span class="co">#&gt;  experiments with user-defined names and respective classes.</span>
<span class="co">#&gt;  Containing an ExperimentList class object of length 2:</span>
<span class="co">#&gt;  [1] scADT: matrix with 13 rows and 8617 columns</span>
<span class="co">#&gt;  [2] scRNAseq: matrix with 36280 rows and 8617 columns</span>
<span class="co">#&gt; Functionality:</span>
<span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span>
<span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span>
<span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span>
<span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span>
<span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="co">#&gt;  exportClass() - save data to flat files</span></code></pre></div>
<p>We see two modalities in the object — <code>scRNAseq</code> and <code>scADT</code>, the latter providing counts for antibody-derived tags. Notably, each experiment is a matrix.</p>
</div>
<div class="section level2">
<h2 id="processing-adt-data">Processing ADT data<a class="anchor" aria-label="anchor" href="#processing-adt-data"></a>
</h2>
<p>While CITE-seq analysis workflows such as <a href="http://www.bioconductor.org/packages/release/bioc/html/CiteFuse.html" class="external-link">CiteFuse</a> should be consulted for more details, below we exemplify simple data transformation in order to demonstrate how their output can be saved to an H5MU file later on.</p>
<p>For ADT counts, we will apply CLR transformation following <a href="https://doi.org/10.1016/j.cell.2021.04.048" class="external-link">Hao et al., 2020</a>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Define CLR transformation as in the Seurat workflow</span>
<span class="va">clr</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">data</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span>
    <span class="va">x</span> <span class="op">/</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="va">x</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span><span class="op">)</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="op">)</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p>We will make the ADT modality a <code>SingleCellExperiment</code> object and add an assay with CLR-transformed counts:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">adt_counts</span> <span class="op">&lt;-</span> <span class="va">mae</span><span class="op">[[</span><span class="st">"scADT"</span><span class="op">]</span><span class="op">]</span>

<span class="va">mae</span><span class="op">[[</span><span class="st">"scADT"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a></span><span class="op">(</span><span class="va">adt_counts</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="st">"scADT"</span><span class="op">]</span><span class="op">]</span>, <span class="st">"clr"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">clr</span><span class="op">(</span><span class="va">adt_counts</span><span class="op">)</span></code></pre></div>
<p>We will also generate reduced dimensions taking advantage of the functionality in the <a href="https://bioconductor.org/packages/release/bioc/vignettes/scater/inst/doc/overview.html" class="external-link"><code>scater</code> package</a>:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mae</span><span class="op">[[</span><span class="st">"scADT"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span>
  <span class="va">mae</span><span class="op">[[</span><span class="st">"scADT"</span><span class="op">]</span><span class="op">]</span>, exprs_values <span class="op">=</span> <span class="st">"clr"</span>, ncomponents <span class="op">=</span> <span class="fl">20</span>
<span class="op">)</span>
<span class="co">#&gt; Warning in check_numbers(k = k, nu = nu, nv = nv, limit = min(dim(x)) - : more</span>
<span class="co">#&gt; singular values/vectors requested than available</span>
<span class="co">#&gt; Warning in (function (A, nv = 5, nu = nv, maxit = 1000, work = nv + 7, reorth =</span>
<span class="co">#&gt; TRUE, : You're computing too large a percentage of total singular values, use a</span>
<span class="co">#&gt; standard svd instead.</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html" class="external-link">plotReducedDim</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="st">"scADT"</span><span class="op">]</span><span class="op">]</span>, dimred <span class="op">=</span> <span class="st">"PCA"</span>,
               by_exprs_values <span class="op">=</span> <span class="st">"clr"</span>, colour_by <span class="op">=</span> <span class="st">"CD3"</span><span class="op">)</span></code></pre></div>
<p><img src="Cord-Blood-CITE-seq_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html" class="external-link">plotReducedDim</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="st">"scADT"</span><span class="op">]</span><span class="op">]</span>, dimred <span class="op">=</span> <span class="st">"PCA"</span>,
               by_exprs_values <span class="op">=</span> <span class="st">"clr"</span>, colour_by <span class="op">=</span> <span class="st">"CD14"</span><span class="op">)</span></code></pre></div>
<p><img src="Cord-Blood-CITE-seq_files/figure-html/unnamed-chunk-7-2.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="writing-h5mu-files">Writing H5MU files<a class="anchor" aria-label="anchor" href="#writing-h5mu-files"></a>
</h2>
<p>We can write the contents of the MultiAssayExperiment object into an H5MU file:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/writeH5MU.html">writeH5MU</a></span><span class="op">(</span><span class="va">mae</span>, <span class="st">"cord_blood_citeseq.h5mu"</span><span class="op">)</span></code></pre></div>
<p>We can check that both modalities were written to the file, whether it was a <code>matrix</code> for RNA or <code>SingleCellExperiment</code> for ADT:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h5</span> <span class="op">&lt;-</span> <span class="fu">rhdf5</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Fopen.html" class="external-link">H5Fopen</a></span><span class="op">(</span><span class="st">"cord_blood_citeseq.h5mu"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html" class="external-link">h5ls</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Gopen.html" class="external-link">H5Gopen</a></span><span class="op">(</span><span class="va">h5</span>, <span class="st">"mod"</span><span class="op">)</span>, recursive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt;   group     name     otype dclass dim</span>
<span class="co">#&gt; 0     /    scADT H5I_GROUP           </span>
<span class="co">#&gt; 1     / scRNAseq H5I_GROUP</span></code></pre></div>
<p>… both assays for ADT — raw counts are stored in <code>X</code> and CLR-transformed counts are in the corresponding layer:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html" class="external-link">h5ls</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Gopen.html" class="external-link">H5Gopen</a></span><span class="op">(</span><span class="va">h5</span>, <span class="st">"mod/scADT"</span><span class="op">)</span>, recursive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt;   group   name       otype  dclass       dim</span>
<span class="co">#&gt; 0     /      X H5I_DATASET INTEGER 13 x 8617</span>
<span class="co">#&gt; 1     / layers   H5I_GROUP                  </span>
<span class="co">#&gt; 2     /    obs   H5I_GROUP                  </span>
<span class="co">#&gt; 3     /   obsm   H5I_GROUP                  </span>
<span class="co">#&gt; 4     /    var   H5I_GROUP</span>
<span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html" class="external-link">h5ls</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Gopen.html" class="external-link">H5Gopen</a></span><span class="op">(</span><span class="va">h5</span>, <span class="st">"mod/scADT/layers"</span><span class="op">)</span>, recursive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt;   group name       otype dclass       dim</span>
<span class="co">#&gt; 0     /  clr H5I_DATASET  FLOAT 13 x 8617</span></code></pre></div>
<p>… as well as reduced dimensions (PCA):</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html" class="external-link">h5ls</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Gopen.html" class="external-link">H5Gopen</a></span><span class="op">(</span><span class="va">h5</span>, <span class="st">"mod/scADT/obsm"</span><span class="op">)</span>, recursive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt;   group name       otype dclass       dim</span>
<span class="co">#&gt; 0     /  PCA H5I_DATASET  FLOAT 12 x 8617</span>
<span class="co"># There is an alternative way to access groups:</span>
<span class="co"># h5&amp;'mod'&amp;'scADT'&amp;'obsm'</span>
<span class="fu">rhdf5</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5functions.html" class="external-link">H5close</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li><p><a href="https://www.biorxiv.org/content/10.1101/2021.06.01.445670" class="external-link">Muon: multimodal omics analysis framework</a> preprint</p></li>
<li><p><a href="https://mudata.readthedocs.io/" class="external-link">mudata</a> (Python) documentation</p></li>
<li><p>muon <a href="https://muon.readthedocs.io/" class="external-link">documentation</a> and <a href="https://gtca.github.io/muon/" class="external-link">web page</a></p></li>
<li><p>Stoeckius, M., Hafemeister, C., Stephenson, W., Houck-Loomis, B., Chattopadhyay, P.K., Swerdlow, H., Satija, R. and Smibert, P., 2017. Simultaneous epitope and transcriptome measurement in single cells. <em>Nature methods</em>, 14(9), pp.865-868.</p></li>
<li><p>Hao, Y., Hao, S., Andersen-Nissen, E., Mauck III, W.M., Zheng, S., Butler, A., Lee, M.J., Wilk, A.J., Darby, C., Zager, M. and Hoffman, P., 2021. Integrated analysis of multimodal single-cell data. <em>Cell</em>.</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R Under development (unstable) (2022-05-30 r82436)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.4 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0</span>
<span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span>
<span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span>
<span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] scater_1.25.1               ggplot2_3.3.6              </span>
<span class="co">#&gt;  [3] scuttle_1.7.2               SingleCellMultiModal_1.9.0 </span>
<span class="co">#&gt;  [5] MultiAssayExperiment_1.23.1 SingleCellExperiment_1.19.0</span>
<span class="co">#&gt;  [7] SummarizedExperiment_1.27.1 Biobase_2.57.1             </span>
<span class="co">#&gt;  [9] GenomicRanges_1.49.0        GenomeInfoDb_1.33.3        </span>
<span class="co">#&gt; [11] IRanges_2.31.0              MatrixGenerics_1.9.0       </span>
<span class="co">#&gt; [13] matrixStats_0.62.0          MuData_0.99.9              </span>
<span class="co">#&gt; [15] rhdf5_2.41.0                S4Vectors_0.35.0           </span>
<span class="co">#&gt; [17] BiocGenerics_0.43.0         Matrix_1.4-1               </span>
<span class="co">#&gt; [19] BiocStyle_2.25.0           </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] jsonlite_1.8.0                magrittr_2.0.3               </span>
<span class="co">#&gt;   [3] ggbeeswarm_0.6.0              magick_2.7.3                 </span>
<span class="co">#&gt;   [5] farver_2.1.0                  rmarkdown_2.14               </span>
<span class="co">#&gt;   [7] fs_1.5.2                      zlibbioc_1.43.0              </span>
<span class="co">#&gt;   [9] ragg_1.2.2                    vctrs_0.4.1                  </span>
<span class="co">#&gt;  [11] memoise_2.0.1                 DelayedMatrixStats_1.19.0    </span>
<span class="co">#&gt;  [13] RCurl_1.98-1.6                htmltools_0.5.2              </span>
<span class="co">#&gt;  [15] AnnotationHub_3.5.0           curl_4.3.2                   </span>
<span class="co">#&gt;  [17] BiocNeighbors_1.15.1          Rhdf5lib_1.19.2              </span>
<span class="co">#&gt;  [19] sass_0.4.1                    bslib_0.3.1                  </span>
<span class="co">#&gt;  [21] desc_1.4.1                    cachem_1.0.6                 </span>
<span class="co">#&gt;  [23] mime_0.12                     lifecycle_1.0.1              </span>
<span class="co">#&gt;  [25] pkgconfig_2.0.3               rsvd_1.0.5                   </span>
<span class="co">#&gt;  [27] R6_2.5.1                      fastmap_1.1.0                </span>
<span class="co">#&gt;  [29] GenomeInfoDbData_1.2.8        shiny_1.7.1                  </span>
<span class="co">#&gt;  [31] digest_0.6.29                 colorspace_2.0-3             </span>
<span class="co">#&gt;  [33] AnnotationDbi_1.59.1          rprojroot_2.0.3              </span>
<span class="co">#&gt;  [35] dqrng_0.3.0                   irlba_2.3.5                  </span>
<span class="co">#&gt;  [37] ExperimentHub_2.5.0           textshaping_0.3.6            </span>
<span class="co">#&gt;  [39] RSQLite_2.2.14                beachmat_2.13.1              </span>
<span class="co">#&gt;  [41] labeling_0.4.2                filelock_1.0.2               </span>
<span class="co">#&gt;  [43] fansi_1.0.3                   httr_1.4.3                   </span>
<span class="co">#&gt;  [45] compiler_4.3.0                bit64_4.0.5                  </span>
<span class="co">#&gt;  [47] withr_2.5.0                   BiocParallel_1.31.6          </span>
<span class="co">#&gt;  [49] viridis_0.6.2                 DBI_1.1.2                    </span>
<span class="co">#&gt;  [51] highr_0.9                     HDF5Array_1.25.0             </span>
<span class="co">#&gt;  [53] R.utils_2.11.0                rappdirs_0.3.3               </span>
<span class="co">#&gt;  [55] DelayedArray_0.23.0           rjson_0.2.21                 </span>
<span class="co">#&gt;  [57] tools_4.3.0                   vipor_0.4.5                  </span>
<span class="co">#&gt;  [59] beeswarm_0.4.0                interactiveDisplayBase_1.35.0</span>
<span class="co">#&gt;  [61] httpuv_1.6.5                  R.oo_1.24.0                  </span>
<span class="co">#&gt;  [63] glue_1.6.2                    rhdf5filters_1.9.0           </span>
<span class="co">#&gt;  [65] promises_1.2.0.1              grid_4.3.0                   </span>
<span class="co">#&gt;  [67] generics_0.1.2                gtable_0.3.0                 </span>
<span class="co">#&gt;  [69] R.methodsS3_1.8.1             BiocSingular_1.13.0          </span>
<span class="co">#&gt;  [71] ScaledMatrix_1.5.0            utf8_1.2.2                   </span>
<span class="co">#&gt;  [73] XVector_0.37.0                ggrepel_0.9.1                </span>
<span class="co">#&gt;  [75] BiocVersion_3.16.0            pillar_1.7.0                 </span>
<span class="co">#&gt;  [77] stringr_1.4.0                 limma_3.53.2                 </span>
<span class="co">#&gt;  [79] later_1.3.0                   dplyr_1.0.9                  </span>
<span class="co">#&gt;  [81] BiocFileCache_2.5.0           lattice_0.20-45              </span>
<span class="co">#&gt;  [83] bit_4.0.4                     tidyselect_1.1.2             </span>
<span class="co">#&gt;  [85] locfit_1.5-9.5                Biostrings_2.65.0            </span>
<span class="co">#&gt;  [87] knitr_1.39                    gridExtra_2.3                </span>
<span class="co">#&gt;  [89] bookdown_0.26                 edgeR_3.39.1                 </span>
<span class="co">#&gt;  [91] xfun_0.31                     DropletUtils_1.17.0          </span>
<span class="co">#&gt;  [93] stringi_1.7.6                 yaml_2.3.5                   </span>
<span class="co">#&gt;  [95] evaluate_0.15                 tibble_3.1.7                 </span>
<span class="co">#&gt;  [97] BiocManager_1.30.18           cli_3.3.0                    </span>
<span class="co">#&gt;  [99] xtable_1.8-4                  systemfonts_1.0.4            </span>
<span class="co">#&gt; [101] munsell_0.5.0                 jquerylib_0.1.4              </span>
<span class="co">#&gt; [103] Rcpp_1.0.8.3                  dbplyr_2.1.1                 </span>
<span class="co">#&gt; [105] png_0.1-7                     parallel_4.3.0               </span>
<span class="co">#&gt; [107] ellipsis_0.3.2                pkgdown_2.0.3                </span>
<span class="co">#&gt; [109] assertthat_0.2.1              blob_1.2.3                   </span>
<span class="co">#&gt; [111] sparseMatrixStats_1.9.0       bitops_1.0-7                 </span>
<span class="co">#&gt; [113] SpatialExperiment_1.7.0       viridisLite_0.4.0            </span>
<span class="co">#&gt; [115] scales_1.2.0                  purrr_0.3.4                  </span>
<span class="co">#&gt; [117] crayon_1.5.1                  rlang_1.0.2                  </span>
<span class="co">#&gt; [119] cowplot_1.1.1                 KEGGREST_1.37.0</span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Danila Bredikhin, Ilia Kats.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
