<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting started with MuData for MultiAssayExperiment • MuData</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting started with MuData for MultiAssayExperiment">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MuData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.11.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Blood-CITE-seq.html">CITE-seq data with CiteFuse and MuData</a></li>
    <li><a class="dropdown-item" href="../articles/Cord-Blood-CITE-seq.html">CITE-seq data with MultiAssayExperiment and MuData</a></li>
    <li><a class="dropdown-item" href="../articles/Getting-Started.html">Getting started with MuData for MultiAssayExperiment</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ilia-kats/MuData/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Getting started with MuData for MultiAssayExperiment</h1>
                        <h4 data-toc-skip class="author">Danila
Bredikhin</h4>
            <address class="author_afil">
      European Molecular Biology Laboratory, Heidelberg,
Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:danila.bredikhin@embl.de" class="email">danila.bredikhin@embl.de</a>
      </address>
                              <h4 data-toc-skip class="author">Ilia
Kats</h4>
            <address class="author_afil">
      German Cancer Research Center, Heidelberg,
Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:i.kats@dkfz-heidelberg.de" class="email">i.kats@dkfz-heidelberg.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">2025-02-24</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ilia-kats/MuData/blob/master/vignettes/Getting-Started.Rmd" class="external-link"><code>vignettes/Getting-Started.Rmd</code></a></small>
      <div class="d-none name"><code>Getting-Started.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Multimodal data format — <a href="https://mudata.readthedocs.io/" class="external-link">MuData</a> — <a href="https://www.biorxiv.org/content/10.1101/2021.06.01.445670v1" class="external-link">has
been introduced</a> to address the need for cross-platform standard for
sharing large-scale multimodal omics data. Importantly, it develops
ideas of and is compatible with <a href="https://anndata.readthedocs.io/" class="external-link">AnnData</a> standard for storing
raw and derived data for unimodal datasets.</p>
<p>In Bioconductor, multimodal datasets have been stored in <a href="https://bioconductor.org/packages/MultiAssayExperiment" class="external-link">MultiAssayExperiment</a>
(MAE) objects. This <code>MuData</code> package provides functionality
to read data from MuData files into MAE objects as well as to save MAE
objects into H5MU files.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The most recent dev build can be installed from GitHub:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://remotes.r-lib.org" class="external-link">remotes</a></span><span class="op">)</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"ilia-kats/MuData"</span><span class="op">)</span></span></code></pre></div>
<p>Stable version of <code>MuData</code> will be available in future
bioconductor versions.</p>
</div>
<div class="section level2">
<h2 id="loading-libraries">Loading libraries<a class="anchor" aria-label="anchor" href="#loading-libraries"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ilia-kats/MuData" class="external-link">MuData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/" class="external-link">MultiAssayExperiment</a></span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/grimbough/rhdf5" class="external-link">rhdf5</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="writing-h5mu-files">Writing H5MU files<a class="anchor" aria-label="anchor" href="#writing-h5mu-files"></a>
</h2>
<p>We’ll use a simple MAE object from the
<code>MultiAssayExperiment</code> package that we’ll then save in a H5MU
file.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">miniACC</span><span class="op">)</span></span>
<span><span class="va">miniACC</span></span>
<span><span class="co">#&gt; A MultiAssayExperiment object of 5 listed</span></span>
<span><span class="co">#&gt;  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">#&gt;  Containing an ExperimentList class object of length 5:</span></span>
<span><span class="co">#&gt;  [1] RNASeq2GeneNorm: SummarizedExperiment with 198 rows and 79 columns</span></span>
<span><span class="co">#&gt;  [2] gistict: SummarizedExperiment with 198 rows and 90 columns</span></span>
<span><span class="co">#&gt;  [3] RPPAArray: SummarizedExperiment with 33 rows and 46 columns</span></span>
<span><span class="co">#&gt;  [4] Mutations: matrix with 97 rows and 90 columns</span></span>
<span><span class="co">#&gt;  [5] miRNASeqGene: SummarizedExperiment with 471 rows and 80 columns</span></span>
<span><span class="co">#&gt; Functionality:</span></span>
<span><span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">#&gt;  exportClass() - save data to flat files</span></span></code></pre></div>
<p>We will now write its contents into an H5MU file with
<code>WriteH5MU</code>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/writeH5MU.html">writeH5MU</a></span><span class="op">(</span><span class="va">miniACC</span>, <span class="st">"miniacc.h5mu"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reading-h5mu-files">Reading H5MU files<a class="anchor" aria-label="anchor" href="#reading-h5mu-files"></a>
</h2>
<p>We can manually check the top level of the structure of the file:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rhdf5</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html" class="external-link">h5ls</a></span><span class="op">(</span><span class="st">"miniacc.h5mu"</span>, recursive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt;   group   name     otype dclass dim</span></span>
<span><span class="co">#&gt; 0     /    mod H5I_GROUP           </span></span>
<span><span class="co">#&gt; 1     /    obs H5I_GROUP           </span></span>
<span><span class="co">#&gt; 2     /   obsm H5I_GROUP           </span></span>
<span><span class="co">#&gt; 3     / obsmap H5I_GROUP           </span></span>
<span><span class="co">#&gt; 4     /    uns H5I_GROUP           </span></span>
<span><span class="co">#&gt; 5     /    var H5I_GROUP</span></span></code></pre></div>
<p>Or dig deeper into the file:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h5</span> <span class="op">&lt;-</span> <span class="fu">rhdf5</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Fopen.html" class="external-link">H5Fopen</a></span><span class="op">(</span><span class="st">"miniacc.h5mu"</span><span class="op">)</span></span>
<span><span class="va">h5</span><span class="op">&amp;</span><span class="st">'mod'</span></span>
<span><span class="co">#&gt; HDF5 GROUP </span></span>
<span><span class="co">#&gt;         name /mod</span></span>
<span><span class="co">#&gt;     filename </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              name     otype dclass dim</span></span>
<span><span class="co">#&gt; 0 Mutations       H5I_GROUP           </span></span>
<span><span class="co">#&gt; 1 RNASeq2GeneNorm H5I_GROUP           </span></span>
<span><span class="co">#&gt; 2 RPPAArray       H5I_GROUP           </span></span>
<span><span class="co">#&gt; 3 gistict         H5I_GROUP           </span></span>
<span><span class="co">#&gt; 4 miRNASeqGene    H5I_GROUP</span></span>
<span><span class="fu">rhdf5</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5functions.html" class="external-link">H5close</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="creating-mae-objects-from-h5mu-files">Creating MAE objects from H5MU files<a class="anchor" aria-label="anchor" href="#creating-mae-objects-from-h5mu-files"></a>
</h3>
<p>This package provides <code>ReadH5MU</code> to create an object with
data from an H5MU file. Since H5MU structure has been designed to
accommodate more structured information than MAE, only some data will be
read. For instance, MAE has no support for loading multimodal embeddings
or pairwise graphs.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">acc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readH5MU.html">readH5MU</a></span><span class="op">(</span><span class="st">"miniacc.h5mu"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Reading as SingleCellExperiment where the original object class is matrix</span></span>
<span><span class="co">#&gt; Warning: sampleMap[['assay']] coerced with as.factor()</span></span>
<span><span class="va">acc</span></span>
<span><span class="co">#&gt; A MultiAssayExperiment object of 5 listed</span></span>
<span><span class="co">#&gt;  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">#&gt;  Containing an ExperimentList class object of length 5:</span></span>
<span><span class="co">#&gt;  [1] RNASeq2GeneNorm: SummarizedExperiment with 198 rows and 79 columns</span></span>
<span><span class="co">#&gt;  [2] gistict: SummarizedExperiment with 198 rows and 90 columns</span></span>
<span><span class="co">#&gt;  [3] RPPAArray: SummarizedExperiment with 33 rows and 46 columns</span></span>
<span><span class="co">#&gt;  [4] Mutations: SingleCellExperiment with 97 rows and 90 columns</span></span>
<span><span class="co">#&gt;  [5] miRNASeqGene: SummarizedExperiment with 471 rows and 80 columns</span></span>
<span><span class="co">#&gt; Functionality:</span></span>
<span><span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">#&gt;  exportClass() - save data to flat files</span></span></code></pre></div>
<p>Importantly, we recover the information from the original MAE
object:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">miniACC</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 6 rows and 4 columns</span></span>
<span><span class="co">#&gt;                 patientID years_to_birth vital_status days_to_death</span></span>
<span><span class="co">#&gt;               &lt;character&gt;      &lt;integer&gt;    &lt;integer&gt;     &lt;integer&gt;</span></span>
<span><span class="co">#&gt; TCGA-OR-A5J1 TCGA-OR-A5J1             58            1          1355</span></span>
<span><span class="co">#&gt; TCGA-OR-A5J2 TCGA-OR-A5J2             44            1          1677</span></span>
<span><span class="co">#&gt; TCGA-OR-A5J3 TCGA-OR-A5J3             23            0            NA</span></span>
<span><span class="co">#&gt; TCGA-OR-A5J4 TCGA-OR-A5J4             23            1           423</span></span>
<span><span class="co">#&gt; TCGA-OR-A5J5 TCGA-OR-A5J5             30            1           365</span></span>
<span><span class="co">#&gt; TCGA-OR-A5J6 TCGA-OR-A5J6             29            0            NA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">acc</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 6 rows and 4 columns</span></span>
<span><span class="co">#&gt;                 patientID years_to_birth vital_status days_to_death</span></span>
<span><span class="co">#&gt;               &lt;character&gt;      &lt;integer&gt;    &lt;integer&gt;       &lt;array&gt;</span></span>
<span><span class="co">#&gt; TCGA-OR-A5J1 TCGA-OR-A5J1             58            1          1355</span></span>
<span><span class="co">#&gt; TCGA-OR-A5J2 TCGA-OR-A5J2             44            1          1677</span></span>
<span><span class="co">#&gt; TCGA-OR-A5J3 TCGA-OR-A5J3             23            0            NA</span></span>
<span><span class="co">#&gt; TCGA-OR-A5J4 TCGA-OR-A5J4             23            1           423</span></span>
<span><span class="co">#&gt; TCGA-OR-A5J5 TCGA-OR-A5J5             30            1           365</span></span>
<span><span class="co">#&gt; TCGA-OR-A5J6 TCGA-OR-A5J6             29            0            NA</span></span></code></pre></div>
<p>Features metadata is also recovered:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">miniACC</span><span class="op">[[</span><span class="st">"gistict"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 6 rows and 3 columns</span></span>
<span><span class="co">#&gt;        Gene.Symbol    Locus.ID    Cytoband</span></span>
<span><span class="co">#&gt;        &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; DIRAS3      DIRAS3        9077      1p31.3</span></span>
<span><span class="co">#&gt; MAPK14      MAPK14        1432     6p21.31</span></span>
<span><span class="co">#&gt; YAP1          YAP1       10413     11q22.1</span></span>
<span><span class="co">#&gt; CDKN1B      CDKN1B        1027     12p13.1</span></span>
<span><span class="co">#&gt; ERBB2        ERBB2        2064       17q12</span></span>
<span><span class="co">#&gt; G6PD          G6PD        2539        Xq28</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">acc</span><span class="op">[[</span><span class="st">"gistict"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 6 rows and 3 columns</span></span>
<span><span class="co">#&gt;        Gene.Symbol    Locus.ID    Cytoband</span></span>
<span><span class="co">#&gt;        &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; DIRAS3      DIRAS3        9077      1p31.3</span></span>
<span><span class="co">#&gt; MAPK14      MAPK14        1432     6p21.31</span></span>
<span><span class="co">#&gt; YAP1          YAP1       10413     11q22.1</span></span>
<span><span class="co">#&gt; CDKN1B      CDKN1B        1027     12p13.1</span></span>
<span><span class="co">#&gt; ERBB2        ERBB2        2064       17q12</span></span>
<span><span class="co">#&gt; G6PD          G6PD        2539        Xq28</span></span></code></pre></div>
<div class="section level4">
<h4 id="backed-objects">Backed objects<a class="anchor" aria-label="anchor" href="#backed-objects"></a>
</h4>
<p>It is possible to read H5MU files while keeping matrices (both
<code>.X</code> and <code>.layers</code>) on disk.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">acc_b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readH5MU.html">readH5MU</a></span><span class="op">(</span><span class="st">"miniacc.h5mu"</span>, backed <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Reading as SingleCellExperiment where the original object class is matrix</span></span>
<span><span class="co">#&gt; Warning: sampleMap[['assay']] coerced with as.factor()</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">acc_b</span>, <span class="st">"RNASeq2GeneNorm"</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt; &lt;5 x 3&gt; DelayedMatrix object of type "double":</span></span>
<span><span class="co">#&gt;        TCGA-OR-A5J1-01A-11R-A29S-07 TCGA-OR-A5J2-01A-11R-A29S-07</span></span>
<span><span class="co">#&gt; DIRAS3                    1487.0317                       9.6631</span></span>
<span><span class="co">#&gt; MAPK14                     778.5783                    2823.6469</span></span>
<span><span class="co">#&gt; YAP1                      1009.6061                    2305.0590</span></span>
<span><span class="co">#&gt; CDKN1B                    2101.3449                    4248.9584</span></span>
<span><span class="co">#&gt; ERBB2                      651.2968                     246.4098</span></span>
<span><span class="co">#&gt;        TCGA-OR-A5J3-01A-11R-A29S-07</span></span>
<span><span class="co">#&gt; DIRAS3                      18.9602</span></span>
<span><span class="co">#&gt; MAPK14                    1061.7686</span></span>
<span><span class="co">#&gt; YAP1                      1561.2502</span></span>
<span><span class="co">#&gt; CDKN1B                    1348.5410</span></span>
<span><span class="co">#&gt; ERBB2                       90.0607</span></span></code></pre></div>
<p>The data in the assay is a <code>DelayedMatrix</code> object:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">acc_b</span>, <span class="st">"RNASeq2GeneNorm"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "DelayedMatrix"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "DelayedArray"</span></span></code></pre></div>
<p>This is in contrast to the <code>acc</code> object that has matrices
in memory:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">acc</span>, <span class="st">"RNASeq2GeneNorm"</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt;        TCGA-OR-A5J1-01A-11R-A29S-07 TCGA-OR-A5J2-01A-11R-A29S-07</span></span>
<span><span class="co">#&gt; DIRAS3                    1487.0317                       9.6631</span></span>
<span><span class="co">#&gt; MAPK14                     778.5783                    2823.6469</span></span>
<span><span class="co">#&gt; YAP1                      1009.6061                    2305.0590</span></span>
<span><span class="co">#&gt; CDKN1B                    2101.3449                    4248.9584</span></span>
<span><span class="co">#&gt; ERBB2                      651.2968                     246.4098</span></span>
<span><span class="co">#&gt;        TCGA-OR-A5J3-01A-11R-A29S-07</span></span>
<span><span class="co">#&gt; DIRAS3                      18.9602</span></span>
<span><span class="co">#&gt; MAPK14                    1061.7686</span></span>
<span><span class="co">#&gt; YAP1                      1561.2502</span></span>
<span><span class="co">#&gt; CDKN1B                    1348.5410</span></span>
<span><span class="co">#&gt; ERBB2                       90.0607</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">acc</span>, <span class="st">"RNASeq2GeneNorm"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "matrix" "array"</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li><p><a href="https://www.biorxiv.org/content/10.1101/2021.06.01.445670" class="external-link">Muon:
multimodal omics analysis framework</a> preprint</p></li>
<li><p><a href="https://mudata.readthedocs.io/" class="external-link">mudata</a> (Python)
documentation</p></li>
<li><p>muon <a href="https://muon.readthedocs.io/" class="external-link">documentation</a> and
<a href="https://gtca.github.io/muon/" class="external-link">web page</a></p></li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R Under development (unstable) (2025-02-22 r87796)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.1 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] MultiAssayExperiment_1.33.9 SummarizedExperiment_1.37.0</span></span>
<span><span class="co">#&gt;  [3] Biobase_2.67.0              GenomicRanges_1.59.1       </span></span>
<span><span class="co">#&gt;  [5] GenomeInfoDb_1.43.4         IRanges_2.41.3             </span></span>
<span><span class="co">#&gt;  [7] MatrixGenerics_1.19.1       matrixStats_1.5.0          </span></span>
<span><span class="co">#&gt;  [9] MuData_1.11.1               rhdf5_2.51.2               </span></span>
<span><span class="co">#&gt; [11] S4Vectors_0.45.4            BiocGenerics_0.53.6        </span></span>
<span><span class="co">#&gt; [13] generics_0.1.3              Matrix_1.7-2               </span></span>
<span><span class="co">#&gt; [15] BiocStyle_2.35.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] sass_0.4.9                  SparseArray_1.7.6          </span></span>
<span><span class="co">#&gt;  [3] lattice_0.22-6              h5mread_0.99.4             </span></span>
<span><span class="co">#&gt;  [5] digest_0.6.37               evaluate_1.0.3             </span></span>
<span><span class="co">#&gt;  [7] grid_4.5.0                  bookdown_0.42              </span></span>
<span><span class="co">#&gt;  [9] fastmap_1.2.0               jsonlite_1.9.0             </span></span>
<span><span class="co">#&gt; [11] BiocManager_1.30.25         httr_1.4.7                 </span></span>
<span><span class="co">#&gt; [13] SingleCellExperiment_1.29.1 UCSC.utils_1.3.1           </span></span>
<span><span class="co">#&gt; [15] HDF5Array_1.35.15           textshaping_1.0.0          </span></span>
<span><span class="co">#&gt; [17] jquerylib_0.1.4             abind_1.4-8                </span></span>
<span><span class="co">#&gt; [19] cli_3.6.4                   rlang_1.1.5                </span></span>
<span><span class="co">#&gt; [21] crayon_1.5.3                XVector_0.47.2             </span></span>
<span><span class="co">#&gt; [23] cachem_1.1.0                DelayedArray_0.33.6        </span></span>
<span><span class="co">#&gt; [25] yaml_2.3.10                 BiocBaseUtils_1.9.0        </span></span>
<span><span class="co">#&gt; [27] S4Arrays_1.7.3              tools_4.5.0                </span></span>
<span><span class="co">#&gt; [29] Rhdf5lib_1.29.0             GenomeInfoDbData_1.2.13    </span></span>
<span><span class="co">#&gt; [31] R6_2.6.1                    lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt; [33] fs_1.6.5                    htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt; [35] ragg_1.3.3                  desc_1.4.3                 </span></span>
<span><span class="co">#&gt; [37] pkgdown_2.1.1               bslib_0.9.0                </span></span>
<span><span class="co">#&gt; [39] systemfonts_1.2.1           xfun_0.51                  </span></span>
<span><span class="co">#&gt; [41] knitr_1.49                  rhdf5filters_1.19.1        </span></span>
<span><span class="co">#&gt; [43] htmltools_0.5.8.1           rmarkdown_2.29             </span></span>
<span><span class="co">#&gt; [45] compiler_4.5.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Danila Bredikhin, Ilia Kats.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
